src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ManagedToNativeVTableMethodGenerator.cs

```diff
@@ -185,6 +185,16 @@ public BlockSyntax GenerateStubBody(int index, ImmutableArray<FunctionPointerUnm
             }
 
             tryStatements.AddRange(statements.NotifyForSuccessfulInvoke);
+
+            // GC.KeepAlive(this);
```

Comment:
Let's reference the `Delegate` marshaller symmetry here. At some point it might be worth it to have some sort of "lifetime" phase.

Reply:
The `NotifyForSuccessfulInvoke` phase is the lifetime phase, but the `this` object doesn't have a true marshaller here at the moment (it's hard-coded into the generator). If we ever move it out into an `IMarshallingGenerator`, then we'll move the keep-alive there too.